packages/manager/apps/web/client/app/domain/anycast/domain-dns-anycast.controller.js

```diff
@@ -1,79 +1,91 @@
-    this.Alerter = Alerter;
-    this.atInternet = atInternet;
-    this.Domain = Domain;
+  constructor($q, WucOrderCartService, SessionService) {
+    this.$q = $q;
+    this.WucOrderCartService = WucOrderCartService;
+    this.SessionService = SessionService;
   }
 
   $onInit() {
-    this.domainId = this.domainName;
-    this.optionName = 'dnsAnycast';
-    this.loading = false;
-    this.loadOptionDetails();
-  }
-
-  loadOptionDetails() {
     this.loading = true;
-    this.error = null;
-
-    return this.Domain.getOptionDetails(this.domainId, this.optionName)
-      .then((data) => {
-        this.optionDetails = data;
-      })
-      .catch(() => {
-        this.error = this.$translate.instant(
-          'domain_configuration_dnsanycast_fail',
+    this.fetchCatalogOffer(this.domainName)
+      .then((offer) => {
+        this.offer = offer;
+        this.offerPrice = find(this.offer.prices, {
+          pricingMode: DNS_ANYCAST_PRICING_MODE,
+          duration: DNS_ANYCAST_DURATION,
+        });
+        return this.fetchNewCart().then((cart) =>
+          this.WucOrderCartService.addProductServiceOptionToCart(
+            cart.cartId,
+            DNS_ANYCAST_SERVICE_TYPE,
+            this.domainName,
+            {
+              duration: this.offerPrice.duration,
+              planCode: this.offer.planCode,
+              pricingMode: this.offerPrice.pricingMode,
+              quantity: this.offerPrice.minimumQuantity,
+            },
+          )
+            .then(() =>
+              this.WucOrderCartService.getCheckoutInformations(cart.cartId),
+            )
+            .then((checkout) => {
+              this.cart = cart;
+              this.checkoutInfos = checkout;
+            }),
         );
       })
+      .catch((error) => {
+        this.error = get(error, 'data.message', error);
+      })
       .finally(() => {
         this.loading = false;
       });
   }
 
-  orderDnsanycast() {
-    this.url = null;
-    this.error = null;
-    this.loading = true;
+  fetchNewCart() {
+    return this.SessionService.getUser().then(({ ovhSubsidiary }) =>
+      this.WucOrderCartService.createNewCart(ovhSubsidiary).then((cart) =>
+        this.WucOrderCartService.assignCart(cart.cartId).then(() => cart),
+      ),
+    );
+  }
 
-    return this.Domain.orderOption(
-      this.domainId,
-      this.optionName,
-      this.optionDetails.duration.duration,
-    )
-      .then((order) => {
-        this.order = order;
-        this.url = order.url;
+  fetchCatalogOffer(serviceName) {
+    return this.WucOrderCartService.getProductServiceOptions(
+      DNS_ANYCAST_SERVICE_TYPE,
+      serviceName,
+    ).then((options) => find(options, { planCode: DNS_ANYCAST_PLANCODE }));
+  }
+
+  performCheckout() {
+    this.loading = true;
+    return this.$q((resolve, reject) => {
+      this.WucOrderCartService.checkoutCart(this.cart.cartId, {
+        autoPayWithPreferredPaymentMethod: !!this.defaultPaymentMethod,
```

Comment:
```suggestion
        autoPayWithPreferredPaymentMethod: this.defaultPaymentMethod,
```

Reply:
@radireddy, the autoPayWithPreferredPaymentMethod variable is a boolean, so that's why we need to enforce boolean here.